Add recover testing to the combined module

GiudGiud opened the pull request:
refs #16857

GiudGiud committed:
Set the original configuration of the incremental beam as a restartable attribute
refs  #16857

GiudGiud committed:
Avoid recover testing in combined module for inputs containing objects which do not support recovering
refs  #16857

GiudGiud committed:
Sync solution invalid on EXEC_FAILED in case we exited a solve loop before any iteration completed

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2854659/), step Docs: sync website on 3ea410d wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/30452/site)

This comment will be updated on new commits.

lindsayad commented on modules/combined/examples/geochem-porous_flow/forge/tests:
```diff
@@ -14,6 +14,8 @@
     requirement = "The system shall be able to solve aqueous geochemical systems in contact with minerals, for water in a reservoir."
     issues = '#15693'
     design = 'forge.md'
+    # Geochemistry solver does not support recover
```
Is there an issue for this?

GiudGiud replied:
https://github.com/idaholab/moose/issues/15816

lindsayad commented on modules/combined/test/tests/heat_conduction_xfem/tests:
```diff
@@ -1,10 +1,12 @@
     # XFEM requires --enable-unique-ids in libmesh
     capabilities = 'unique_id'
-  [../]
+    # XFEM does not support recover
```
is there an issue for this?

GiudGiud replied:
https://github.com/idaholab/moose/issues/11567 is probably the closest. Want a dedicated one?

lindsayad replied:
Nah I think that one is good. Referencing the issue number in the comment would be nice

GiudGiud committed:
Address review:
- add issue number to 'does not support recover' comments

GiudGiud committed:
Set recover =false for a test that properly errors already with:
The AbaqusUMATStress umat was stored in restart but no longer exists. This is not supported when recovering stateful material properties.
